packages/core/admin/server/src/controllers/authentication.ts

```diff
@@ -10,23 +12,111 @@ import {
   validateResetPasswordInput,
-  validateRenewTokenInput,
+  validateLoginSessionInput,
 } from '../validation/authentication';
 
 import type {
   ForgotPassword,
   Login,
   Register,
   RegistrationInfo,
-  RenewToken,
   ResetPassword,
 } from '../../../shared/contracts/authentication';
 import { AdminUser } from '../../../shared/contracts/shared';
 
 const { ApplicationError, ValidationError } = errors;
 
+const refreshCookieName = 'strapi_admin_refresh';
+
+const getRefreshCookieOptions = () => {
+  const isProduction = strapi.config.get('environment') === 'production';
+  const domain: string | undefined =
+    strapi.config.get('admin.auth.cookie.domain') || strapi.config.get('admin.auth.domain');
+  const path: string = strapi.config.get('admin.auth.cookie.path', '/admin');
+
+  const sameSite: boolean | 'lax' | 'strict' | 'none' =
+    strapi.config.get('admin.auth.cookie.sameSite') ?? 'lax';
+
+  return {
+    httpOnly: true,
+    secure: isProduction,
+    overwrite: true,
+    domain,
+    path,
+    sameSite,
+    maxAge: undefined,
+  };
+};
+
+const getLifespansForFamilyType = (
+  familyType: 'refresh' | 'session'
+): { idleSeconds: number; maxSeconds: number } => {
+  if (familyType === 'refresh') {
+    const idleSeconds = Number(
+      strapi.config.get('admin.auth.sessions.idleRefreshTokenLifespan', 7 * 24 * 60 * 60)
+    );
+    const maxSeconds = Number(
+      strapi.config.get('admin.auth.sessions.maxRefreshTokenLifespan', 30 * 24 * 60 * 60)
+    );
+
+    return { idleSeconds, maxSeconds };
+  }
+
+  const idleSeconds = Number(strapi.config.get('admin.auth.sessions.idleSessionLifespan', 60 * 60));
+  const maxSeconds = Number(
+    strapi.config.get('admin.auth.sessions.maxSessionLifespan', 7 * 24 * 60 * 60)
+  );
+
+  return { idleSeconds, maxSeconds };
+};
+
+/**
+ * Builds cookie options applying an expiration policy based on family type.
+ * - refresh (remember me): persistent cookie; Expires is min(now+idle, absoluteMax)
+ * - session (non-remember): session cookie; no Expires/Max-Age
+ */
+const buildCookieOptionsWithExpiry = (
+  familyType: 'refresh' | 'session',
+  absoluteExpiresAtISO?: string
+) => {
+  const base = getRefreshCookieOptions();
+  if (familyType === 'session') {
+    return base;
+  }
+
+  const { idleSeconds } = getLifespansForFamilyType('refresh');
+  const now = Date.now();
+  const idleExpiry = now + idleSeconds * 1000;
+  const absoluteExpiry = absoluteExpiresAtISO
+    ? new Date(absoluteExpiresAtISO).getTime()
+    : idleExpiry;
+  const chosen = new Date(Math.min(idleExpiry, absoluteExpiry));
+
+  return { ...base, expires: chosen, maxAge: Math.max(0, chosen.getTime() - now) };
+};
+
+const getSessionManager = (): Modules.SessionManager.SessionManagerService | null => {
+  const manager = strapi.sessionManager as Modules.SessionManager.SessionManagerService | undefined;
+
+  return manager ?? null;
+};
+
+const generateDeviceId = (): string => crypto.randomUUID();
```

Comment:
Do we need a fallback for this in the server code? The admin should always
provide it